src/napari_epitools/_widget.py

```diff
@@ -173,10 +173,33 @@ def run() -> npt.NDArray[np.float64]:
     return run()
 
 
+def _init_segmentation_widget(widget):
+    """Add callbacks for the widget"""
+
+    viewer = napari.current_viewer()
+
+    # Automatically select a newly added Image layer
+    viewer.layers.events.inserted.connect(
+        lambda event: _select_inserted_image(event.value, widget.input_image),
+    )
+
+
+def _select_inserted_image(new_layer, widget):
+    """Update the selected Image when a image layer is added"""
+
+    if not isinstance(new_layer, napari.layers.Image):
+        return
```

Comment:
Should we log something here?

Reply:
I don't think we have any logging set up yet. Shall we open another issue to set up logging for everything (including this, [the cell statistics widget](https://github.com/epitools/epitools/pull/44), and the current code)?

Reply:
I've opened #46 to set up logging

Reply:
Sounds like a reasonable compromise